synapse/storage/databases/main/user_directory.py

```diff
@@ -234,10 +234,10 @@ def _get_next_batch(
         processed_event_count = 0
 
         for room_id, event_count in rooms_to_work_on:
-            is_in_room = await self.is_host_joined(room_id, self.server_name)
+            is_in_room = await self.is_host_joined(room_id, self.server_name)  # type: ignore[attr-defined]
```

Comment:
Hmm, I'm quite uneasy about how many `# type: ignore[attr-defined]` there are here. If we're not careful we'll end up losing a lot of useful typing info. I'm not sure what the best way forward here is though, will canvas for opinions

Reply:
Consensus was this was better than nothing

Reply:
I don't like it that much, but better to "ignore" the few lines and check the rest of the file.